tests/compiler/test_typing.py

```diff
@@ -143,18 +145,72 @@ def test_cast_to_number(snippetcompiler):
     root: Namespace = scopes.get_child("__config__")
     a = root.lookup("a").get_value()
     b = root.lookup("b").get_value()
+    c = root.lookup("b").get_value()
     x = root.lookup("x").get_value()
     y = root.lookup("y").get_value()
     z = root.lookup("z").get_value()
     u = root.lookup("u").get_value()
     assert Number().validate(a)
     assert Number().validate(b)
+    assert Number().validate(c)
     assert Number().validate(x)
     assert Number().validate(y)
     assert Number().validate(z)
     assert Number().validate(u)
 
 
+def test_compare_float_int_attribute(snippetcompiler):
+    snippetcompiler.setup_for_snippet(
+        """
+    entity Test:
+        float i = 0
+    end
+
+    entity Test2:
+        int i = 0
+    end
+
+    implement Test using std::none
+    implement Test2 using std::none
+
+    val1 = Test(i = 42.0)
+    val2 = Test2(i = 42)
+    val2.i = val1.i
+    val1.i = val2.i
```

Comment:
This is not a good test because the value is already set, therefore we fall into an execution flow where the validation is not called. I would expect this to fail.

Reply:
I'll replace it with another test